### PR TITLE
Refactoring and fixes in net_ws

### PR DESCRIPTION
**Pull request by In-line**

### CODE REVIEW

File: rehlds/engine/net_ws.cpp [hunk 1]
```diff
@@ -349,106 +351,63 @@ char *NET_BaseAdrToString(netadr_t& a)
...
+			
+			// Convert from hexademical represantation to sockaddr
+			char temp[3] = { s[i], s[i + 1], '\0' };
+			sscanf(temp, "%x", &val);
+			sadr->sa_data[i / 2] = (char)val;
+		}
+		
 		sscanf(s + 22, "%u", &val);
-		*(uint16 *)&sadr->sa_data[10] = htons(val);
+-		*(uint16 *)&sadr->sa_data[10] = htons(val);
```
Comment (theAsmodai) [inline rehlds/engine/net_ws.cpp]: what about sockaddr_in6?

File: rehlds/engine/net_ws.cpp [hunk 2]
```diff
@@ -349,106 +351,63 @@ char *NET_BaseAdrToString(netadr_t& a)
...
+	if(colon != nullptr)
 	{
-		if (*colon == ':')
-		{
-			*colon = 0;
-			val = Q_atoi(colon + 1);
-			((sockaddr_in *)sadr)->sin_port = htons(val);
-		}
-		colon++;
+		sadr_in->sin_port = htons(Q_atoi(colon + 1));
```
Comment (theAsmodai) [inline rehlds/engine/net_ws.cpp]: `*colon = '\0'` was missed

File: rehlds/engine/net_ws.cpp [hunk 3]
```diff
@@ -1898,6 +1819,15 @@ int NET_IsConfigured()
 	return net_configured;
 }
 
+bool NET_CheckPort(int port)
+{
+#ifdef REHLDS_FIXES
+	return port > 0 && port <= 65535;
```
Comment (theAsmodai) [inline rehlds/engine/net_ws.cpp]: USHRT_MAX?

File: rehlds/engine/net_ws.cpp [hunk 4]
```diff
@@ -624,12 +577,9 @@ void NET_RemoveFromPacketList(packetlag_t *pPacket)
...
-
-	int c;
-	packetlag_t *p;
-
-	c = 0;
-	p = pList->pNext;
+	
+	int c = 0;
+	packetlag_t *p = pList->pNext;
 	while (p && p != pList)
```
Comment (WPMGPRoSToTeMa) [inline rehlds/engine/net_ws.cpp]: Maybe `for (packetlag_t* p = pList->pNext; p && p != pList; p = p->pNext)`?

**Comment by In-line** [comment]
Thanks for all suggestions!

File: rehlds/engine/net_ws.cpp (hunk 1 above)
Comment (In-line) [reply rehlds/engine/net_ws.cpp]: @theAsmodai Maybe add ipv6 support in another commit?

**Comment by In-line** [comment]
Finally I fixed build for Windows

**Pull request merged**